File: 34-find-first-and-last-position-of-element-in-sorted-array/34-find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
    
    int BSearch(vector<int>& nums, int target, bool firstOccurence){
        int res = -1;
        int l = 0;
        int r = nums.size() - 1;
        
        while(l <= r){
            int mid = l + (r - l) / 2;
            if(nums[mid] == target){
                res = mid;
                if(firstOccurence)
                    r = mid - 1;
                else
                    l = mid + 1;
            }
            
            else if(nums[mid] > target)
                r = mid - 1;
            else
                l = mid + 1;
        }
        
        return res;
    }
    
    
    
    vector<int> searchRange(vector<int>& nums, int target) {  
        return {BSearch(nums,target,true),BSearch(nums,target,false)};
    }
};

File: 34-find-first-and-last-position-of-element-in-sorted-array/34-find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        
        
        int[] res = {-1,-1};
        if(nums == null || nums.length == 0)
            return res;
        
        
        int i=0 , j=nums.length-1, mid=i + (j - i) / 2;
        while(i<=j){
            mid = i + (j - i) / 2;
            if(nums[mid] == target) break;
            else if(nums[mid] < target) i = mid+1;
            else j = mid-1;
        }
        
        System.out.print(nums[mid]);
        
        if(nums[mid] != target){
            return res;
        }
            
        i = mid; j=mid;
        while(j<nums.length && nums[j] == target)
            j++;
        while(i>=0 && nums[i] == target)
            i--;
        
        res[0] = i+1;
        res[1] = j-1;
        return res;
        
    }
}